code/_core/obj/item/weapon/_weapon.dm

```diff
@@ -18,6 +18,8 @@
 
 	var/enchantment/enchantment
 
+	var/list/enchantment_whitelist[] //A whitelist of [/enchantment/path] or [ALL] of enchants allowed on the weapon E.G. /enchantment/fire.
```

Comment:
The [] should be removed. Also this should be a blacklist given that blacklisting a group of gems seems more likely.

Reply:
The [] was an accident, so thanks for pointing that out. I defend the whitelist with the fact that when a weapon is added, it should be designed with everything in mind, by making it a whitelist, you ensure that that item is able to handle that enchantment, where as a blacklist is for saying it cant work with that enchantment. Will add a blacklist to it.

Reply:
Blacklist added